post-processing/test_post_processing.py

```diff
@@ -237,6 +228,22 @@ def test_high_level_script(run_sombrero):
     else:
         assert False
 
+    # check correct display name parsing
+    try:
+        df = post_.run_post_processing(sombrero_changed_log_path, {"filters": [], "series": [], "x_axis": {"value": "tasks", "units": {"custom": None}}, "y_axis": {"value": "cpus_per_task", "units": {"column": "extra_param"}}})
+    except Exception as e:
+        # three param columns found in changed log
+        EXPECTED_FIELDS = ["tasks", "cpus_per_task", "extra_param"]
+        # FIXME: assert a certain column dtype for the extra param here?
+        assert e.args[1].columns.tolist() == EXPECTED_FIELDS
+    else:
+        assert False
+
+    # check correct date filtering
+    df = post_.run_post_processing(sombrero_changed_log_path, {"title": "Title", "filters": [["job_completion_time", ">", "2000-06-01T12:30:15"]], "series": [], "x_axis": {"value": "job_completion_time", "units": {"custom": None}}, "y_axis": {"value": "flops_value", "units": {"column": "flops_unit"}}})
+    # check returned subset is as expected
+    assert len(df) == 2
+

```

Comment:
Add a test that compares `datetime` with `date`, and is expected to fail with type error. In a new issue, we should fix this by converting any occurence of `date` in user-specified filters or series to `datetime`.

Reply:
I've discovered an additional complication; if the filter value of your date is interpreted as a string (i.e. you use something like `"2000-06-01"` in your config file), the filter comparison goes ahead and filters your data as expected. However, if the filter value of your date is `datetime.date` (i.e. you use `"2000-06-01"` in your config file), then you get a `TypeError`.